src/ApiService/ApiService/onefuzzlib/NotificationOperations.cs

```diff
@@ -0,0 +1,154 @@
+using System.Text.Json;
+using ApiService.OneFuzzLib.Orm;
+
+namespace Microsoft.OneFuzz.Service;
+
+public interface INotificationOperations
+{
+    Async.Task NewFiles(Container container, string filename, bool failTaskOnTransientError);
+}
+
+public class NotificationOperations : Orm<Notification>, INotificationOperations
+{
+    private ILogTracer _log;
+    private IReports _reports;
+    private ITaskOperations _taskOperations;
+
+    private IContainers _containers;
+
+    private IQueue _queue;
+
+    private IEvents _events;
+
+    public NotificationOperations(ILogTracer log, IStorage storage, IReports reports, ITaskOperations taskOperations, IContainers containers, IQueue queue, IEvents events)
+        : base(storage)
+    {
+        _log = log;
+        _reports = reports;
+        _taskOperations = taskOperations;
+        _containers = containers;
+        _queue = queue;
+        _events = events;
+    }
+    public async Async.Task NewFiles(Container container, string filename, bool failTaskOnTransientError)
+    {
+        var notifications = GetNotifications(container);
+        var hasNotifications = await notifications.AnyAsync();
+        var report = await _reports.GetReportOrRegression(container, filename, expectReports: hasNotifications);
+
+        if (!hasNotifications)
+        {
+            return;
+        }
+
+        var done = new List<NotificationTemplate>();
+        await foreach (var notification in notifications)
+        {
+            if (done.Contains(notification.Config))
+            {
+                continue;
+            }
+
+            done.Add(notification.Config);
+
+            if (notification.Config.TeamsTemplate != null)
+            {
+                NotifyTeams(notification.Config.TeamsTemplate, container, filename, report);
+            }
+
+            if (report == null)
+            {
+                continue;
+            }
+
+            if (notification.Config.AdoTemplate != null)
+            {
+                NotifyAdo(notification.Config.AdoTemplate, container, filename, report, failTaskOnTransientError);
+            }
+
+            if (notification.Config.GithubIssuesTemplate != null)
+            {
+                GithubIssue(notification.Config.GithubIssuesTemplate, container, filename, report);
+            }
+        }
+
+        await foreach (var (task, containers) in GetQueueTasks())
+        {
+            if (containers.Contains(container.ContainerName))
+            {
+                _log.Info($"queuing input {container.ContainerName} {filename} {task.TaskId}");
+                var url = _containers.GetFileSasUrl(container, filename, StorageType.Corpus, read: true, delete: true);
+                await _queue.SendMessage(task.TaskId.ToString(), System.Text.Encoding.UTF8.GetBytes(url.ToString()), StorageType.Corpus);
```

Comment:
It would be nice to have an override for _queue.SendMessage that took a string instead of bytes and did the conversion internally